cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1519,335 +1519,372 @@ function Start_Containerd {
+    while ((Get-service $service).Status -ne 'Stopped' -and $waited -lt $timeout) {
       Start-Sleep 1
       $waited++
 
       if ($waited % $log_period -eq 0) {
-        Log-Output "Waiting for StackdriverLogging service to stop"
+        Log-Output "Waiting for ${service} service to stop"
       }
     }
 
     # Timeout occurred
     if ($waited -ge $timeout) {
-      Throw ("Timeout while waiting for StackdriverLogging service to stop")
+      Throw ("Timeout while waiting for ${service} service to stop")
     }
   }
 
-  Start-Service StackdriverLogging
+  Start-Service $service
 }
 
 # Check whether the logging agent is installed by whether it's registered as service
 function IsLoggingAgentInstalled {
-  $stackdriver_status = (Get-Service StackdriverLogging -ErrorAction Ignore).Status
-  return -not [string]::IsNullOrEmpty($stackdriver_status)
+  $logging_status = (Get-Service $LOGGINGAGENT_SERVICE -ErrorAction Ignore).Status
+  return -not [string]::IsNullOrEmpty($logging_status)
 }
 
 # Clean up the logging agent's registry key and root folder if they exist from a prior installation.
 # Try to uninstall it first, if it failed, remove the registry key at least, 
 # as the registry key will block the silent installation later on.
 function Cleanup-LoggingAgent {
-  # For 64 bits app, the registry path is 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall'
-  # for 32 bits app, it's 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall'
-  # StackdriverLogging is installed as 32 bits app
-  $x32_app_reg = 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall'
-  $uninstall_string = (Get-ChildItem $x32_app_reg | Get-ItemProperty | Where-Object {$_.DisplayName -match "Stackdriver"}).UninstallString
-  if (-not [string]::IsNullOrEmpty($uninstall_string)) {
-    try {
-      Start-Process -FilePath "$uninstall_string" -ArgumentList "/S" -Wait
-    } catch {
-      Log-Output "Exception happens during uninstall logging agent, so remove the registry key at least"
-      Remove-Item -Path "$x32_app_reg\GoogleStackdriverLoggingAgent\"
-    }
-  }
-
-  #  If we chose reboot after uninstallation, the root folder would be clean.
-  #  But since we couldn't reboot, so some files & folders would be left there, 
-  #  which could block the re-installation later on, so clean it up
-  if(Test-Path $STACKDRIVER_ROOT){
-    Remove-Item -Force -Recurse $STACKDRIVER_ROOT
-  }
+  Log-Output ("Skip: Cleanup logging agent TODO")
 }
 
-# Installs the Stackdriver logging agent according to
-# https://cloud.google.com/logging/docs/agent/installation.
-# TODO(yujuhong): Update to a newer Stackdriver agent once it is released to
-# support kubernetes metadata properly. The current version does not recognizes
-# the local resource key "logging.googleapis.com/local_resource_id", and fails
-# to label namespace, pod and container names on the logs.
+# Installs the logging agent according to
```

Comment:
Incomplete comment. Please link to where these installation steps are documented.